relay-event-normalization/src/normalize/processor.rs

```diff
@@ -0,0 +1,1495 @@
+use std::collections::hash_map::DefaultHasher;
+use std::collections::BTreeSet;
+
+use std::hash::{Hash, Hasher};
+use std::mem;
+
+use chrono::{DateTime, Duration, Utc};
+use relay_base_schema::metrics::{is_valid_metric_name, DurationUnit, FractionUnit, MetricUnit};
+use relay_event_schema::processor::{
+    self, MaxChars, ProcessingAction, ProcessingResult, ProcessingState, Processor,
+};
+use relay_event_schema::protocol::{
+    AsPair, Context, ContextInner, Contexts, DeviceClass, Event, EventType, Exception, Headers,
+    IpAddr, LogEntry, Measurement, Measurements, Request, SpanAttribute, SpanStatus, Tags,
+    TraceContext, User,
+};
+use relay_protocol::{Annotated, Empty, Error, ErrorKind, Meta, Object, Value};
+use smallvec::SmallVec;
+
+use crate::normalize::{mechanism, stacktrace};
+use crate::span::tag_extraction::{self, extract_span_tags};
+use crate::timestamp::TimestampProcessor;
+use crate::utils::MAX_DURATION_MOBILE_MS;
+use crate::{
+    breakdowns, schema, span, transactions, trimming, user_agent, BreakdownsConfig,
+    ClockDriftProcessor, DynamicMeasurementsConfig, GeoIpLookup, RawUserAgentInfo,
+};
+
+use crate::LightNormalizationConfig;
+
+/// Configuration for [`NormalizeProcessor`].
+#[derive(Debug, Default)]
+pub struct NormalizeProcessorConfig<'a> {
+    /// Light normalization config.
+    // XXX(iker): we should split this config appropriately.
+    light_config: LightNormalizationConfig<'a>,
```

Comment:
To have this indirection with few layers of configs is really petty.  
Can this be already flattened into the `NormalizeProcessorConfig` in this PR?

Reply:
I'll keep the light normalization config around for now, and move keys out of it as we get rid of it. Eventually, it'll have no keys and we'll remove it.